includes/migrate_historic_events_to_current_hidden_pii_configuration.js

```diff
@@ -58,6 +76,38 @@ WHERE
   AND DATE(occurred_at) <= CURRENT_DATE - 1 ;
 `;}).join('\n')}
 
+${["`" + params.bqProjectName + "." + params.bqDatasetName + "." + params.bqEventsTableName + "`", ctx.ref("events_" + params.eventSourceName)].map(eventsTableReference => {return `
+UPDATE
+  ${eventsTableReference} event
+SET
+  data = ARRAY(SELECT AS STRUCT data_combined.key, data_combined.value FROM UNNEST(ARRAY_CONCAT(data, hidden_data)) data_combined WHERE data_combined.key IN UNNEST(entity.visible_keys)),
+  hidden_data = ARRAY(SELECT AS STRUCT data_combined.key, data_combined.value FROM UNNEST(ARRAY_CONCAT(data, hidden_data)) data_combined WHERE data_combined.key IN UNNEST(entity.hidden_keys) OR key NOT IN UNNEST(entity.visible_keys))
+FROM (
+  SELECT
+    event_type,
+    ARRAY(SELECT DISTINCT key FROM UNNEST(keys) WHERE configured_to_be_hidden_in_data_schema IS TRUE) AS hidden_keys,
+    ARRAY(SELECT DISTINCT key FROM UNNEST(keys) WHERE configured_to_be_hidden_in_data_schema IS FALSE) AS visible_keys
+  FROM
+  UNNEST([
+      ${params.customEventSchema.map(customEvent => {
+    return `STRUCT("${customEvent.eventType}" AS event_type,
+        [
+            ${customEvent.keys.filter(key => !(key.historic)).map(key => { return `STRUCT("${key.keyName}" AS key, ${key.hidden || 'false'} AS configured_to_be_hidden_in_data_schema)`; }).join(', ')}
+        ] AS keys
+    )`;
+  }
+  ).join(',')}  
+  ])
+) AS entity

```

Comment:
Is it worth renaming entity here to custom_event, for consistency with the rest of the code?

Reply:
Good idea - done